src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -183,30 +199,58 @@ public void onOpen(Session session, EndpointConfig config) throws IOException {
     }
 
     /**
-     * Notify listeners on each event received on the WebSocket.
+     * Notify listeners on each event received on the websocket and buffer partial messages.
      *
-     * @param message The message received on this WebSocket
+     * @param partialMessage partial message received on this websocket
+     * @param last indicate the last part of a message
+     * @throws MessageTooBigException in case the message is longer than maxMessageLength
      */
     @OnMessage
-    public void onMessage(String message) {
-        if (message != null && !message.equals("server received message")) {
-            // Salt API adds a "data: " prefix that we need to ignore
-            Event event = JsonParser.EVENTS.parse(message.substring(6));
-            synchronized (listeners) {
-                listeners.stream().forEach(l -> l.notify(event));
+    public void onMessage(String partialMessage, boolean last)
+            throws MessageTooBigException {
+        if (last) {
+            String message;
+            if (partialMessageBuffer.length() == 0) {
+                message = partialMessage;
+            } else {
+                partialMessageBuffer.append(partialMessage);
+                message = partialMessageBuffer.toString();
+
+                // Reset the size to the defaultBufferSize and empty the buffer
+                partialMessageBuffer.setLength(defaultBufferSize);
+                partialMessageBuffer.trimToSize();
+                partialMessageBuffer.setLength(0);
+            }
+            if (maxMessageLength > 0 && message.length() > maxMessageLength) {
+                throw new MessageTooBigException(maxMessageLength);
+            }
+
+            // Notify all registered listeners
+            if (!message.equals("server received message")) {
+                // Salt API adds a "data: " prefix that we need to ignore
+                Event event = JsonParser.EVENTS.parse(message.substring(6));
+                synchronized (listeners) {
+                    listeners.stream().forEach(listener -> listener.notify(event));
+                }
+            }
+        } else {
+            partialMessageBuffer.append(partialMessage);
+            if (maxMessageLength > 0 && partialMessageBuffer.length() > maxMessageLength) {
```

Comment:
No, you can't do that because maxMessageLength is special. The check for >0 needs to remain.

    maxMessageLength = config.get(ClientConfig.WEBSOCKET_MAX_MESSAGE_LENGTH);

if it is <= 0, then max length is to be ignored.